starlite/testing.py

```diff
@@ -280,6 +281,356 @@ def test_my_handler() -> None:
+
+        Args:
+            headers: A dictionary of headers.
+            cookies: A string representing the cookie header or a list of "Cookie" instances.
+                This value can include multiple cookies.
+        Returns:
+            A list of encoded headers that can be passed to the request scope.
+        """
+
+        headers = headers or {}
+        self._create_cookie_header(headers, cookies)
+        return [
+            ((key.lower()).encode("latin-1", errors="ignore"), value.encode("latin-1", errors="ignore"))
+            for key, value in headers.items()
+        ]
+
+    def _create_request_with_data(
+        self,
+        http_method: HttpMethod,
+        path: str,
+        headers: Optional[Dict[str, str]] = None,
+        cookies: Optional[Union[List["Cookie"], str]] = None,
+        user: Any = None,
+        auth: Any = None,
+        request_media_type: RequestEncodingType = RequestEncodingType.JSON,
+        data: Optional[Union[Dict[str, Any], "BaseModel"]] = None,
+    ) -> Request[Any, Any]:
+        """Create a [Request][starlite.connection.Request] instance that has
+        body (data)
+
+        Args:
+            http_method: The request's HTTP method.
+            path: The request's path.
+            headers: A dictionary of headers.
+            cookies: A string representing the cookie header or a list of "Cookie" instances.
+                This value can include multiple cookies.
+            user: A value for `request.scope["user"]`
+            auth: A value for `request.scope["auth"]`
+            request_media_type: The 'Content-Type' header of the request.
+            data: A value for the request's body. Can be either a pydantic model instance
+                or a string keyed dictionary.
+        Returns:
+            A [Request][starlite.connection.Request] instance
+        """
+
+        scope = self._create_scope(path, http_method, user, auth)
+
+        headers = headers or {}
+        if data:
+            if isinstance(data, BaseModel):
+                data = data.dict()
+            if request_media_type == RequestEncodingType.JSON:
+                body = dumps(data)
+                headers["Content-Type"] = str(RequestEncodingType.JSON.value)
+            elif request_media_type == RequestEncodingType.MULTI_PART:
+                body, content_type = RequestEncoder().multipart_encode(data)
+                headers["Content-Type"] = content_type
+            else:
+                body = RequestEncoder().url_encode(data)
+                headers["Content-Type"] = str(RequestEncodingType.URL_ENCODED.value)
+            scope["_body"] = body
+
+        self._create_cookie_header(headers, cookies)
+        scope["headers"] = self._build_headers(headers)
+        return Request(scope=scope)  # type: ignore[arg-type]
+
+    def get(
+        self,
+        path: str,
+        headers: Optional[Dict[str, str]] = None,
+        cookies: Optional[Union[List["Cookie"], str]] = None,
```

Comment:
Also add `session` as an argument here and in rest of the methods. It will take dictionary of type `Dict[str, Any]` as value from the user. Then assign this value to `scope['session']`.

Reply:
`session` added here: **https://github.com/starlite-api/starlite/pull/497/commits/d871e917dc768b2bd00ffe2c346e00cb281d647f**